src/main/java/com/csforge/sstable/Query.java

```diff
@@ -99,6 +106,16 @@ public Query(String query, Collection<File> path, CFMetaData cfm) throws Illegal
             filter = builder.build();
         }
         this.queriedColumns = filter;
+
+        try {
+            // Make getAggregationSpecification public and invoke it..
+            Method getAggregationSpecification = SelectStatement.RawStatement.class.getDeclaredMethod("getAggregationSpecification", CFMetaData.class, Selection.class, StatementRestrictions.class, boolean.class);
```

Comment:
Without providing an aggregationSpecification, `ResultSetBuilder.newRow` isn't able to appropriately determine when and when not to create a new row result.   For example `select count(*)` started returning a row for every row in the sstable, instead of just 1.  This also seems to enable `group by` aggregations as well.